src/context_menu.rs

```diff
@@ -48,3 +65,149 @@ impl ContextMenu {
         }
     }
 }
+
+/// Context Menu
+#[cfg(linux)]
+#[derive(Debug, Clone)]
+pub struct ContextMenu {
+    menu_items: Vec<MenuItem>,
+    /// The webview that the context menu is attached to
+    webview: WebView,
+}
+
+#[cfg(linux)]
+impl ContextMenu {
+    /// Create a dialog in the window.
+    ///
+    /// Often used by calling window.alert() or window.confirm() in the web page.
+    pub fn new_with_menu(menu_items: Vec<MenuItem>) -> Self {
+        let webview_id = WebViewId::new();
+        let webview = WebView::new(webview_id, DeviceIntRect::zero());
+
+        Self {
+            menu_items,
+            webview,
+        }
+    }
+
+    /// Show the context menu to current cursor position
+    pub fn show(
+        &mut self,
+        sender: &Sender<ConstellationMsg>,
+        window: &mut Window,
+        position: PhysicalPosition<f64>,
+    ) {
+        let scale_factor = window.scale_factor();
+        self.set_position(window, position, scale_factor);
+
+        send_to_constellation(
+            sender,
+            ConstellationMsg::NewWebView(self.resource_url(), self.webview.webview_id),
+        );
+        window.append_dialog_webview(self.webview.clone());
```

Comment:
I am unsure what you mean, but I pushed a new commit that modified it to check the condition by window.context_menu.webview instead of dialog_webviews now.